src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -801,17 +803,38 @@ public override int GetHashCode()
 
         public bool Equals(Guid g) => EqualsCore(this, g);
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static bool EqualsCore(in Guid left, in Guid right)
         {
-            ref int rA = ref Unsafe.AsRef(in left._a);
-            ref int rB = ref Unsafe.AsRef(in right._a);
+            if (Sse2.IsSupported)
+            {
+                var g1 = Unsafe.As<Guid, Vector128<byte>>(ref Unsafe.AsRef(left));
```

Comment:
Note: use explicit types, not `var`, to follow the coding guidelines in this repo.

Reply:
I have updated accordingly.